### PR TITLE
Add fluxcd integration

### PR DESCRIPTION
**Pull request by melchiormoulin**
### What does this PR do?
Add [fluxcd](https://fluxcd.io/) integration
This integration add an openmetric v2 check
As the first step to bring observability flux metrics to the community, then add dashboards and monitors later. 

### Motivation
Flux is a set of continuous and progressive delivery solutions for Kubernetes that are open and extensible.
Flux is a CNCF Incubating project.

What inspired you to submit this pull request?
We use flux for CD and we want to bring an integration to the Datadog community.

### Review checklist

- [X] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [X] Feature or bugfix has tests
- [X] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by melchiormoulin** [comment]
how can i run again the CI please ?

**Comment by melchiormoulin** [comment]
`ddev validate all fluxcd` and `ddev test` are ok on my laptop.

File: fluxcd/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+The fluxcd integration does not include any service checks.
+
+See [service_checks.json][8] for a list of service checks provided by this integration.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][9].
+
+
+[1]: **LINK_TO_INTEGRATION_SITE**
```
Comment (hestonhoffman) [inline fluxcd/README.md:59]:
```suggestion
[1]: https://fluxcd.io/
```
Is this a good URL?

File: fluxcd/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+[2]: https://app.datadoghq.com/account/settings#agent
+[3]: https://docs.datadoghq.com/agent/kubernetes/integrations/
+[4]: https://github.com/DataDog/integrations-core/blob/master/check/datadog_checks/check/data/conf.yaml.example
```
Comment (hestonhoffman) [inline fluxcd/README.md:62]:
```suggestion
[4]: https://github.com/DataDog/integrations-extras/blob/master/neutrona/datadog_checks/fluxcd/data/conf.yaml.example
```

File: fluxcd/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Agent Check: fluxcd
+
+## Overview
+[Flux](https://fluxcd.io/) is a set of continuous and progressive delivery solutions for Kubernetes that are open and extensible.
+This check monitors [fluxcd][1] through the Datadog Agent.
```
Comment (hestonhoffman) [inline fluxcd/README.md:5]:
```suggestion
This check monitors fluxcd through the Datadog Agent.
```

File: fluxcd/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.
+
+### Installation
+
+To install the fluxcd check on your host:
+
+
+1. Install the [developer toolkit]
+(https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit)
```
Comment (hestonhoffman) [inline fluxcd/README.md:17]:
```suggestion
1. Install the [developer toolkit][10]
```

File: fluxcd/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+[1]: **LINK_TO_INTEGRATION_SITE**
+[2]: https://app.datadoghq.com/account/settings#agent
+[3]: https://docs.datadoghq.com/agent/kubernetes/integrations/
+[4]: https://github.com/DataDog/integrations-core/blob/master/check/datadog_checks/check/data/conf.yaml.example
+[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
+[6]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
+[7]: https://github.com/DataDog/integrations-core/blob/master/check/metadata.csv
+[8]: https://github.com/DataDog/integrations-core/blob/master/check/assets/service_checks.json
+[9]: https://docs.datadoghq.com/help/
```
Comment (hestonhoffman) [inline fluxcd/README.md:67]:
```suggestion
[9]: https://docs.datadoghq.com/help/
[10]: https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit
```

File: fluxcd/README.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+### Events
+
+The fluxcd integration does not include any events.
+
+### Service Checks
+
+The fluxcd integration does not include any service checks.
+
+See [service_checks.json][8] for a list of service checks provided by this integration.
```
Comment (hestonhoffman) [inline fluxcd/README.md:52]:
```suggestion
```

File: fluxcd/README.md (hunk 1 above)
Comment (hestonhoffman) [inline fluxcd/README.md:59]:
```suggestion
```

File: fluxcd/metadata.csv [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+fluxcd.controller.runtime.active.workers,gauge,,,,Number of currently used workers per controller,0,fluxcd,controller runtime active workers,
+fluxcd.controller.runtime.reconcile.count,count,,,, Total number of reconciliations per controller,0,fluxcd,controller runtime reconcile total,
+fluxcd.controller.runtime.reconcile.time.seconds.bucket,count,,,,Bucket of length of time per reconciliation per controller,0,fluxcd,controller runtime reconcile time seconds,
+fluxcd.controller.runtime.reconcile.time.seconds.sum,count,,,,Sum of length of time per reconciliation per controller,0,fluxcd,controller runtime reconcile time seconds,
+fluxcd.controller.runtime.reconcile.time.seconds.count,count,,,,Count of length of time per reconciliation per controller,0,fluxcd,controller runtime reconcile time seconds,
+fluxcd.controller.runtime.max.concurrent.reconciles,gauge,,,,Maximum number of concurrent reconciles per controller,0,fluxcd,controller runtime max concurrent reconciles,
+fluxcd.controller.runtime.reconcile.errors.count,count,,,,Total number of reconciliation errors per controller,0,fluxcd,controller runtime reconcile errors total,
```
Comment (hestonhoffman) [inline fluxcd/metadata.csv:13]:
```suggestion
metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
fluxcd.gotk.reconcile.condition,gauge,,,, The current condition status of a GitOps Toolkit resource reconciliation.,0,fluxcd,gotk reconcile condition,
fluxcd.gotk.suspend.status,gauge,,,,The current suspend status of a GitOps Toolkit resource.,0,fluxcd,gotk suspend status,
fluxcd.gotk.reconcile.duration.seconds.sum,count,,,,Sum of the duration in seconds of a GitOps Toolkit resource reconciliation.,0,fluxcd,reconcile duration seconds,
fluxcd.gotk.reconcile.duration.seconds.count,count,,,,Count of the duration in seconds of a GitOps Toolkit resource reconciliation.,0,fluxcd,reconcile duration seconds,
fluxcd.gotk.reconcile.duration.seconds.bucket,count,,,,Bucket of the duration in seconds of a GitOps Toolkit resource reconciliation.,0,fluxcd,reconcile duration seconds,
fluxcd.controller.runtime.active.workers,gauge,,,,Number of currently used workers per controller.,0,fluxcd,controller runtime active workers,
fluxcd.controller.runtime.reconcile.count,count,,,, Total number of reconciliations per controller.,0,fluxcd,controller runtime reconcile total,
fluxcd.controller.runtime.reconcile.time.seconds.bucket,count,,,,Bucket of length of time per reconciliation per controller.,0,fluxcd,controller runtime reconcile time seconds,
fluxcd.controller.runtime.reconcile.time.seconds.sum,count,,,,Sum of length of time per reconciliation per controller.,0,fluxcd,controller runtime reconcile time seconds,
fluxcd.controller.runtime.reconcile.time.seconds.count,count,,,,Count of length of time per reconciliation per controller.,0,fluxcd,controller runtime reconcile time seconds,
fluxcd.controller.runtime.max.concurrent.reconciles,gauge,,,,Maximum number of concurrent reconciles per controller.,0,fluxcd,controller runtime max concurrent reconciles,
fluxcd.controller.runtime.reconcile.errors.count,count,,,,Total number of reconciliation errors per controller.,0,fluxcd,controller runtime reconcile errors total,
```

File: fluxcd/README.md (hunk 2 above)
Comment (melchiormoulin) [reply fluxcd/README.md:62]:
```suggestion
[4]: https://github.com/DataDog/integrations-extras/blob/master/fluxcd/datadog_checks/fluxcd/data/conf.yaml.example
```

**Comment by jose-manuel-almaza** [comment]
Conflicts must be resolved

**Comment by melchiormoulin** [comment]
> Conflicts must be resolved

done

**Review by jose-manuel-almaza — APPROVED** [review]

**Comment by jose-manuel-almaza** [comment]
Looks good for me

**Comment by melchiormoulin** [comment]
@hestonhoffman all good for you too?

**Review by hestonhoffman — APPROVED** [review]
LGTM!

**Comment by melchiormoulin** [comment]
Hello. Do you know when it can be merged ?

**Comment by jose-manuel-almaza** [comment]
You have the approval. You can merge

**Comment by melchiormoulin** [comment]
i can't, that's why i'm asking.

**Pull request merged**